subsys/bluetooth/mesh/crypto_mtls.c

```diff
@@ -0,0 +1,644 @@
+
+	sg[0].data = out;
+	sg[0].len  = sizeof(out);
+	pad = 0x02;
+
+	err = bt_mesh_aes_cmac(t, sg, ARRAY_SIZE(sg), out);
+	if (err) {
+		return err;
+	}
+
+	memcpy(enc_key, out, 16);
+
+	pad = 0x03;
+
+	err = bt_mesh_aes_cmac(t, sg, ARRAY_SIZE(sg), out);
+	if (err) {
+		return err;
+	}
+
+	memcpy(priv_key, out, 16);
+
+	BT_DBG("NID 0x%02x enc_key %s", net_id[0], bt_hex(enc_key, 16));
+	BT_DBG("priv_key %s", bt_hex(priv_key, 16));
+
+	return 0;
+}
+
+int bt_mesh_k3(const uint8_t n[16], uint8_t out[8])
+{
+	uint8_t id64[] = { 'i', 'd', '6', '4', 0x01 };
+	uint8_t tmp[16];
+	uint8_t t[16];
+	int err;
+
+	err = bt_mesh_s1("smk3", tmp);
+	if (err) {
+		return err;
+	}
+
+	err = bt_mesh_aes_cmac_one(tmp, n, 16, t);
+	if (err) {
+		return err;
+	}
+
+	err = bt_mesh_aes_cmac_one(t, id64, sizeof(id64), tmp);
+	if (err) {
+		return err;
+	}
+
+	memcpy(out, tmp + 8, 8);
+
+	return 0;
+}
+
+int bt_mesh_k4(const uint8_t n[16], uint8_t out[1])
+{
+	uint8_t id6[] = { 'i', 'd', '6', 0x01 };
+	uint8_t tmp[16];
+	uint8_t t[16];
+	int err;
+
+	err = bt_mesh_s1("smk4", tmp);
+	if (err) {
+		return err;
+	}
+
+	err = bt_mesh_aes_cmac_one(tmp, n, 16, t);
+	if (err) {
+		return err;
+	}
+
+	err = bt_mesh_aes_cmac_one(t, id6, sizeof(id6), tmp);
+	if (err) {
+		return err;
+	}
+
+	out[0] = tmp[15] & BIT_MASK(6);
+
+	return 0;
+}
+
+int bt_mesh_id128(const uint8_t n[16], const char *s, uint8_t out[16])
+{
+	const char *id128 = "id128\x01";
+	uint8_t salt[16];
+	int err;
+
+	err = bt_mesh_s1(s, salt);
+	if (err) {
+		return err;
+	}
+
+	return bt_mesh_k1(n, 16, salt, id128, out);
+}
+
+static void create_proxy_nonce(uint8_t nonce[13], const uint8_t *pdu,
+			       uint32_t iv_index)
+{
+	/* Nonce Type */
+	nonce[0] = 0x03;
+
+	/* Pad */
+	nonce[1] = 0x00;
+
+	/* Sequence Number */
+	nonce[2] = pdu[2];
+	nonce[3] = pdu[3];
+	nonce[4] = pdu[4];
+
+	/* Source Address */
+	nonce[5] = pdu[5];
+	nonce[6] = pdu[6];
+
+	/* Pad */
+	nonce[7] = 0U;
+	nonce[8] = 0U;
+
+	/* IV Index */
+	sys_put_be32(iv_index, &nonce[9]);
+}
+
+static void create_net_nonce(uint8_t nonce[13], const uint8_t *pdu,
+			     uint32_t iv_index)
+{
+	/* Nonce Type */
+	nonce[0] = 0x00;
+
+	/* FRND + TTL */
+	nonce[1] = pdu[1];
+
+	/* Sequence Number */
+	nonce[2] = pdu[2];
+	nonce[3] = pdu[3];
+	nonce[4] = pdu[4];
+
+	/* Source Address */
+	nonce[5] = pdu[5];
+	nonce[6] = pdu[6];
+
+	/* Pad */
+	nonce[7] = 0U;
+	nonce[8] = 0U;
+
+	/* IV Index */
+	sys_put_be32(iv_index, &nonce[9]);
+}
+
+int bt_mesh_net_obfuscate(uint8_t *pdu, uint32_t iv_index,
+			  const uint8_t privacy_key[16])
+{
+	uint8_t priv_rand[16] = { 0x00, 0x00, 0x00, 0x00, 0x00, };
+	uint8_t tmp[16];
+	int err, i;
+
+	BT_DBG("IVIndex %u, PrivacyKey %s", iv_index, bt_hex(privacy_key, 16));
+
+	sys_put_be32(iv_index, &priv_rand[5]);
+	memcpy(&priv_rand[9], &pdu[7], 7);
+
+	BT_DBG("PrivacyRandom %s", bt_hex(priv_rand, 16));
+
+	err = bt_encrypt_be(privacy_key, priv_rand, tmp);
+	if (err) {
+		return err;
+	}
+
+	for (i = 0; i < 6; i++) {
```

Comment:
```suggestion
	for (int i = 0; i < 6; i++) {
```

Reply:
the same reason regarding refactoring here as well.